src/screens/NewProfile/NewProfile.js

```diff
@@ -142,32 +139,28 @@ const NewProfile = ({
       && !usernameValidationErrorMessage
       && isUsernameInputDirty) {
       setIsCheckingUsername(true);
-    } else if (usernameValidationErrorMessage && isCheckingUsername) {
-      // reset
+    } else if (isCheckingUsername && (usernameValidationErrorMessage || errorMessage)) {
+      // reset if error occurred during update
       setIsCheckingUsername(false);
     }
   }, [usernameValue]);
 
+  useEffect(() => {
+    // user updated, reset
+    if (isCheckingUsername) setIsCheckingUsername(false);
+  }, [user]);
+
   useEffect(() => {
     if (!usernameValidationErrorMessage) onValidUsername();
     return onValidUsername.cancel;
   }, [onValidUsername, usernameValue]);
 
-
-  useEffect(() => {
-    if (walletState !== CHECKING_USERNAME && isCheckingUsername) setIsCheckingUsername(false);
-  }, [walletState]);
-
   const colors = getThemeColors(theme);
 
-  const existingUser = !!apiUser?.walletId;
+  const existingUser = !!user?.walletId;
```

Comment:
is `existingUser` worth a `useState`?

Reply:
It will change every time `user` prop changes since `user` prop comes from reducer, component level state management for this not needed.

Reply:
Cool 👍